### DISCUSSION TITLE
Using Signals in Lit

### DISCUSSION THREAD

**Discussion by ardic** [lit/lit#3725] [General]
Hello guys and girls,  would it be possible to use Signals / Effects / Memos in  Lit  as in Solid.js to manage the state and to be able to use global states between different web components? Is there any strategy devised for this?

**Comment by usergenic** [comment]
This playground shows a possible approach to leverage Signals with Lit.  It is using the `@preact/signals-core` package.

https://lit.dev/playground/#gist=660427ac6065bed390c438b0eb5b5036

**Reply by jaredcwhite** [reply]
This is seriously legit. Love it!

**Reply by bgotink** [reply]
Here's an alternative approach using a directive to only update the part of the template that needs updating: https://lit.dev/playground/#gist=3f54bef71de4498fe954b70a425dc9c2

**Reply by ardic** [reply]
Thanks for your time @usergenic, very useful. It would be nice if Lit integrated it into its next version so as not to have to carry third-party libraries, since it seems that "this is the way" (as the Mandalorians would say).
Very good contribution @bgotink thanks!!

**Reply by drk-mtr** [reply]
Interesting :) With bgotink's version I hit a stumbling block, because I don't know how to add the global `counter` value to the local `count` value within the component and have the resulting value update when either one of them changes.

**Reply by bgotink** [reply]
@drk-mtr I've updated the playground link above with an example of how this can be done using the `computed` function.

**Reply by drk-mtr** [reply]
Nice, thanks! I have no idea why I didn't try this variation, seems obvious now ha.

**Reply by dannyfritz** [reply]
What would be the recommended way to use the Signal value in a conditional in the template? [playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNpZ25hbC1leGFtcGxlLnRzIiwiY29udGVudCI6ImltcG9ydCB7aHRtbCwgY3NzLCBMaXRFbGVtZW50fSBmcm9tICdsaXQnO1xuaW1wb3J0IHtjdXN0b21FbGVtZW50LCBwcm9wZXJ0eX0gZnJvbSAnbGl0L2RlY29yYXRvcnMuanMnO1xuaW1wb3J0IHt3YXRjaH0gZnJvbSAnLi9zaWduYWwtd2F0Y2hlci5qcyc7XG5pbXBvcnQge3NpZ25hbCwgY29tcHV0ZWR9IGZyb20gJ0BwcmVhY3Qvc2lnbmFscy1jb3JlJztcblxuY29uc3QgY291bnRlciA9IHNpZ25hbCgwKTtcblxuQGN1c3RvbUVsZW1lbnQoJ3NpZ25hbC1leGFtcGxlJylcbmV4cG9ydCBjbGFzcyBTaWduYWxFeGFtcGxlIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIHN0YXRpYyBzdHlsZXMgPSBjc3NgXG4gICAgOmhvc3Qge1xuICAgICAgZGlzcGxheTogYmxvY2s7XG4gICAgICBib3JkZXI6IHNvbGlkIDFweCBibHVlO1xuICAgICAgcGFkZGluZzogOHB4O1xuICAgICAgbWFyZ2luOiA0cHg7XG4gICAgfVxuICBgO1xuXG4gIEBwcm9wZXJ0eSgpIG5hbWU7XG4gIFxuICByZW5kZXJDb3VudCA9IDA7XG5cbiAgcmVuZGVyKCkge1xuICAgIGNvbnN0IHJlbmRlckNvdW50ID0gKyt0aGlzLnJlbmRlckNvdW50O1xuICAgIHJldHVybiBodG1sYFxuICAgICAgPHA-XG4gICAgICAgIFNpZ25hbCBjb3VudGVyOiAke3dhdGNoKGNvdW50ZXIpfVxuICAgICAgICA8YnV0dG9uIEBjbGljaz0keygpID0-IGNvdW50ZXIudmFsdWUrK30-Kys8L2J1dHRvbj5cbiAgICAgIDwvcD5cbiAgICAgIDxwPlxuICAgICAgICAke3dhdGNoKGNvdW50ZXIpICUgMiA9PT0gMCA_IFwiRVZFTlwiIDogXCJPRERcIn1cbiAgICAgIDwvcD5cbiAgICBgO1xuICB9XG59XG4ifSx7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhRE9DVFlQRSBodG1sPlxuPGh0bWw-XG4gIDxoZWFkPlxuICAgIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vc2lnbmFsLWV4YW1wbGUuanNcIj48L3NjcmlwdD5cbiAgICA8c3R5bGU-XG4gICAgICBib2R5IHtcbiAgICAgICAgZm9udC1zaXplOiAxLjI1cmVtO1xuICAgICAgICBmb250LWZhbWlseTogc2Fucy1zZXJpZjtcbiAgICAgIH1cbiAgICA8L3N0eWxlPlxuICA8L2hlYWQ-XG4gIDxib2R5PlxuICAgIDxwPlxuICAgICAgVGhpcyBleGFtcGxlcyBzaG93cyBvZmYgaW50ZWdyYXRpb24gd2l0aFxuICAgICAgPGEgaHJlZj1cImh0dHBzOi8vcHJlYWN0anMuY29tL2Jsb2cvaW50cm9kdWNpbmctc2lnbmFscy9cIj48Y29kZT5AcHJlYWN0L3NpZ25hbHM8L2NvZGU-PC9hPlxuICAgICAgd2l0aCBhIDxjb2RlPlNpZ25hbFdhdGNoZXI8L2NvZGU-IG1peGluIHRoYXQgd2lsbCB1cGRhdGUgdGhlIExpdCBlbGVtZW50XG4gICAgICB3aGVuZXZlciBhbnkgc2lnbmFscyBpdCB1c2VzIGNoYW5nZS5cbiAgICA8L3A-XG4gICAgPHNpZ25hbC1leGFtcGxlIG5hbWU9XCJTaWduYWwgdmFsdWUgaW4gY29uZGl0aW9uYWxcIj48L3NpZ25hbC1leGFtcGxlPlxuICA8L2JvZHk-XG48L2h0bWw-In0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMS4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4yLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9LHsibmFtZSI6InNpZ25hbC13YXRjaGVyLnRzIiwiY29udGVudCI6ImltcG9ydCB7bm9DaGFuZ2V9IGZyb20gJ2xpdCc7XG5pbXBvcnQge2RpcmVjdGl2ZSwgQXN5bmNEaXJlY3RpdmV9IGZyb20gJ2xpdC9hc3luYy1kaXJlY3RpdmUuanMnO1xuaW1wb3J0IHt0eXBlIFNpZ25hbCwgZWZmZWN0fSBmcm9tICdAcHJlYWN0L3NpZ25hbHMtY29yZSc7XG5cbmNsYXNzIFdhdGNoRGlyZWN0aXZlIGV4dGVuZHMgQXN5bmNEaXJlY3RpdmUge1xuICAjY2xlYW51cD86ICgpID0-IHZvaWQ7XG4gICNzaWduYWw_OiBTaWduYWw8dW5rbm93bj47XG5cbiAgcmVuZGVyKHNpZ25hbDogU2lnbmFsPHVua25vd24-KSB7XG4gICAgdGhpcy4jc2lnbmFsID0gc2lnbmFsO1xuXG4gICAgaWYgKHRoaXMuaXNDb25uZWN0ZWQpIHtcbiAgICAgIHRoaXMuI2NsZWFudXA_LigpO1xuICAgICAgdGhpcy4jY2xlYW51cCA9IGVmZmVjdCgoKSA9PiB0aGlzLnNldFZhbHVlKHNpZ25hbC52YWx1ZSkpO1xuICAgIH1cblxuICAgIHJldHVybiBub0NoYW5nZTtcbiAgfVxuXG4gIGRpc2Nvbm5lY3RlZCgpIHtcbiAgICB0aGlzLiNjbGVhbnVwPy4oKTtcbiAgICB0aGlzLiNjbGVhbnVwID0gdW5kZWZpbmVkO1xuICB9XG5cbiAgcmVjb25uZWN0ZWQoKSB7XG4gICAgY29uc3Qgc2lnbmFsID0gdGhpcy4jc2lnbmFsO1xuICAgIGlmIChzaWduYWwpIHtcbiAgICAgIHRoaXMuI2NsZWFudXAgPSBlZmZlY3QoKCkgPT4gdGhpcy5zZXRWYWx1ZShzaWduYWwudmFsdWUpKTtcbiAgICB9XG4gIH1cbn1cblxuZXhwb3J0IGNvbnN0IHdhdGNoID0gZGlyZWN0aXZlKFdhdGNoRGlyZWN0aXZlKTtcbiJ9XQ)

**Reply by dannyfritz** [reply]
Actually, using the SignalWatcher() class version works. Interesting how it uses the open `effect()` to setup the dependencies.

**Comment by e111077** [comment]
Update since @usergenic's post, we are working on an official integration:

https://github.com/lit/lit/discussions/4115

Please continue all further feedback and discussion over on that issue
